app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java

```diff
@@ -123,6 +126,29 @@ public static String getLineWithoutMarkdown(@NonNull String content, int lineNum
         return line;
     }
 
+    /**
+     * Strips all lines beginning with a filled checkbox from the Markdown.
+     *
+     * @param content    String
+     * @return newContent String
+     */
+    @NonNull
+    public static String getContentWithoutCheckedItems(@NonNull String content) {
+        final Pattern pattern = Pattern.compile(
+                "^ *[-+*] \\[[xX]\\].*$",
+                Pattern.MULTILINE
+        );
```

Comment:
This omits some variants that could lead to undefined behavior, for example checkboxes within code blocks. You can get some inspiration how to handle codeblocks in the [`nextcloud-commons`](https://github.com/stefan-niedermann/nextcloud-commons/blob/master/markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java#L140) library, but beware: It ain't that easy. Codeblocks can have three or more backticks, for example

Reply:
Thanks, I will!